cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -289,13 +301,20 @@ func generateMountsModeRun() error {
 			tryKernel, tryKernelExists, err := ebl.TryKernel()
 			// TODO:UC20: can we log somewhere if err != nil here?
 			if tryKernelExists && err == nil {
```

Comment:
this could be just:

`if tryKernelExists {`

?

Reply:
well we could have a confusing situation where `tryKernelExists == true` and `err != nil`, which seems silly, but since this is an interface with many implementations I'd rather be more careful here than more efficient. 

(as an aside, I did actually run into situations where that happened while testing with the mock bootloader implementation because that implementation currently doesn't set the try kernel properly with `EnableTryKernel` for `TryKernel` to return - this bug will be fixed in a PR I am submitting today however)

Reply:
that honestly that would be a bug in TryKernel and  sounds dangerous.  if err != nil then tryKernelExists should be always false. Either way if we want to be paranoid then that would be much less confusing written as:

`if err == nil && tryKernelExists`

but is very surprising for readers that it needs to be that way.

Reply:
I will re-write this as your suggestion with `err == nil` first and also add some comments on the ExtractedRunKernelImageBootloader interface about expected behavior of this method and others

Reply:
hmm as I try to write the doc methods for the ExtractedRunKernelImageBootloader, it appears they could be contentious, will do this in a followup

Reply:
contentious in which way? either way is probably a good idea not to block this on that

Reply:
contentious insofar as I'm not sure how specific we should be in the doc-comments, i.e. should EnableTryKernel have this level of specificity in the doc-comment:

```go
// EnableTryKernel enables the specified kernel on ubuntu-boot to be tried by the bootloader
// on a reboot, to be used in conjunction with setting "kernel_status" to "try". This will usually
// be implemented with a "try-kernel.efi" symlink pointing to the extracted kernel image.
```

Reply:
ah, in that, sense, it's not clear there will always be symlinks so we should be a bit vague on that aspect (we can still mention it as one known way)